diff -ruN avifile-0.46.orig/samples/benchmark/Makefile.in avifile-0.46/samples/benchmark/Makefile.in
--- avifile-0.46.orig/samples/benchmark/Makefile.in	Sun Aug 20 05:15:16 2000
+++ avifile-0.46/samples/benchmark/Makefile.in	Mon Aug 21 20:11:25 2000
@@ -7,7 +7,7 @@
 
 MOC=@MOC@
 LIBS=@LIBS@ -lX11 -lXext -lqt -laviplay $(DGA_LDADD) $(ESD_LDADD) @SDL_LIBS@
-LIB_DIRS=$(QT_LDADD) -L/usr/X11R6/lib -L../../lib
+LIB_DIRS=$(QT_LDADD) -L../../lib -L/usr/X11R6/lib
 CDOPT=@CD_OPT@
 LDOPT=@LD_OPT@
 ESD_LDADD=@ESD_LDADD@
diff -ruN avifile-0.46.orig/samples/ctest/Makefile.in avifile-0.46/samples/ctest/Makefile.in
--- avifile-0.46.orig/samples/ctest/Makefile.in	Sat Aug 19 09:46:14 2000
+++ avifile-0.46/samples/ctest/Makefile.in	Mon Aug 21 20:12:14 2000
@@ -1,6 +1,6 @@
 CFLAGS=-g -I../../include
 CPPFLAGS=-g -I../../include
-LIBS=@X_LDFLAGS@ @LIBS@ -lloader -L../../lib
+LIBS=-L../../lib @X_LDFLAGS@ @LIBS@ -lloader
 
 OBJS=test.o comptest.o
 
diff -ruN avifile-0.46.orig/samples/extractor/Makefile.in avifile-0.46/samples/extractor/Makefile.in
--- avifile-0.46.orig/samples/extractor/Makefile.in	Sat Aug 19 09:46:14 2000
+++ avifile-0.46/samples/extractor/Makefile.in	Mon Aug 21 20:12:29 2000
@@ -1,5 +1,5 @@
 CPPFLAGS=-g -I../../include
-LIBS=@X_LDFLAGS@ @LIBS@ -laviplay -L../../lib
+LIBS=-L../../lib @X_LDFLAGS@ @LIBS@ -laviplay
 
 OBJS=extractor.o
 
diff -ruN avifile-0.46.orig/samples/plustest/Makefile.in avifile-0.46/samples/plustest/Makefile.in
--- avifile-0.46.orig/samples/plustest/Makefile.in	Sat Aug 19 09:46:14 2000
+++ avifile-0.46/samples/plustest/Makefile.in	Mon Aug 21 20:12:46 2000
@@ -1,6 +1,6 @@
 CPPFLAGS=-g -I../../include
 OPTFLAGS=-I..
-LIBS=@X_LDFLAGS@ @LIBS@ -lloader -lvideocodec -L../../lib
+LIBS=-L../../lib @X_LDFLAGS@ @LIBS@ -lloader -lvideocodec
 
 OBJS=main.o
 
