--- avifile-0.7-0.7.45/plugins/libffmpeg/FFAudioDecoder.cpp.orig	2006-12-21 19:14:53.000000000 +0000
+++ avifile-0.7-0.7.45/plugins/libffmpeg/FFAudioDecoder.cpp	2006-12-21 19:15:23.000000000 +0000
@@ -37,7 +37,7 @@
 
 	if (m_pFormat->cbSize > 0)
 	{
-	    m_pAvContext->extradata = (char*)(m_pFormat + 1);
+	    m_pAvContext->extradata = (uint8_t*)(m_pFormat + 1);
 	    m_pAvContext->extradata_size = m_pFormat->cbSize;
 	}
 
--- avifile-0.7-0.7.45/plugins/libffmpeg/FFVideoDecoder.cpp.orig	2006-12-21 19:18:08.000000000 +0000
+++ avifile-0.7-0.7.45/plugins/libffmpeg/FFVideoDecoder.cpp	2006-12-21 19:19:09.000000000 +0000
@@ -197,7 +197,7 @@
 	   )
 	{
             m_pAvContext->extradata_size = m_pFormat->biSize - sizeof(BITMAPINFOHEADER);
-	    m_pAvContext->extradata = (char*) m_pFormat + sizeof(BITMAPINFOHEADER);
+	    m_pAvContext->extradata = (uint8_t*) m_pFormat + sizeof(BITMAPINFOHEADER);
 	    if (m_pAvContext->extradata_size > 40)
 		m_pAvContext->flags |= CODEC_FLAG_EXTERN_HUFF; // somewhat useless
 	}
