diff -Nuard avifile-0.7-0.7.38.orig/configure.in avifile-0.7-0.7.38/configure.in
--- avifile-0.7-0.7.38.orig/configure.in	2003-07-10 13:15:54.000000000 +0200
+++ avifile-0.7-0.7.38/configure.in	2003-10-18 12:19:11.000000000 +0200
@@ -17,7 +17,7 @@
 configure_options="$@"
 echo "avifile configure options: $configure_options"
 
-AC_CONFIG_AUX_DIR(admin)
+#AC_CONFIG_AUX_DIR(admin)
 
 # libtool versioning
 #
@@ -994,7 +995,6 @@
 
 AC_OUTPUT(
  Makefile
- admin/Makefile
  ffmpeg/Makefile
  ffmpeg/libavformat/Makefile
  ffmpeg/libavcodec/Makefile
diff -Nuard avifile-0.7-0.7.38.orig/Makefile.am avifile-0.7-0.7.38/Makefile.am
--- avifile-0.7-0.7.38.orig/Makefile.am	2003-07-10 15:17:59.000000000 +0200
+++ avifile-0.7-0.7.38/Makefile.am	2003-10-17 20:03:18.000000000 +0200
@@ -1,4 +1,4 @@
-SUBDIRS = . admin m4 ffmpeg drivers lib plugins bin \
+SUBDIRS = m4 ffmpeg drivers lib plugins bin \
  doc include debian libavqt player samples
 
 EXTRA_DIST = INSTALL autogen.sh avifile.spec.in avifile.pc.in \
