--- avifile-0.50/samples/qtvidcap/Makefile.am~	Wed Dec  6 17:02:08 2000
+++ avifile-0.50/samples/qtvidcap/Makefile.am	Wed Dec  6 17:12:36 2000
@@ -7,7 +7,7 @@
 qtvidcap_LDADD=../../lib/libaviplay.la capwindow.o codecconf.o vidconf.o
 
 install-exec-local:
-	@INSTALL@ -m4755 ./kv4lsetup @bindir@
+	@INSTALL@ -m4755 ./kv4lsetup $(DESTDIR)@bindir@
 
 CXXFLAGS=-I../../include @X_INCLUDES@ @QT_INCLUDES@ @SDL_CFLAGS@
 
--- avifile-0.50/Makefile.am~	Wed Dec  6 17:14:08 2000
+++ avifile-0.50/Makefile.am	Wed Dec  6 17:24:05 2000
@@ -2,4 +2,4 @@
 noinst_SCRIPTS=README
 DIST_SUDIRS = doc
 install-exec-local:
-	$(INSTALL_PROGRAM) -m0755 ./avifile-config $(bindir)
+	$(INSTALL_PROGRAM) -m0755 ./avifile-config $(DESTDIR)$(bindir)
