--- avifile-0.7-0.7.45/configure.in.orig	2024-11-17 22:12:31.962552188 +0100
+++ avifile-0.7-0.7.45/configure.in	2024-11-17 22:18:05.884647746 +0100
@@ -908,10 +908,6 @@ if test x$enable_release != xyes -o x$en
     dnl CFLAGS="-g $PROFFLAG"
     dnl CXXFLAGS="-g $PROFFLAG"
     dnl FFMPEG_CFLAGS="-g $PROFFLAG"
-else
-    CFLAGS=`echo $CFLAGS | sed s/-g//`
-    CXXFLAGS=`echo $CXXFLAGS | sed s/-g//`
-    FFMPEG_CFLAGS=`echo $FFMPEG_CFLAGS | sed s/-g//`
 fi
 
 WIN32_CFLAGS=`echo $CFLAGS | sed s/-fomit-frame-pointer//g`
