diff -urN avifile-0.6.org/Makefile.am avifile-0.6/Makefile.am
--- avifile-0.6.org/Makefile.am	Fri Mar 23 17:27:09 2001
+++ avifile-0.6/Makefile.am	Fri Mar 23 17:27:53 2001
@@ -1,4 +1,4 @@
-BASE_SUBDIRS = include plugins lib bin doc
+BASE_SUBDIRS = include lib plugins bin doc
 
 if AM_USE_QT
 QT_SUBDIRS = player samples
diff -urN avifile-0.6.org/plugins/libaudiodec/Makefile.am avifile-0.6/plugins/libaudiodec/Makefile.am
--- avifile-0.6.org/plugins/libaudiodec/Makefile.am	Fri Mar 23 17:27:10 2001
+++ avifile-0.6/plugins/libaudiodec/Makefile.am	Fri Mar 23 17:27:20 2001
@@ -7,4 +7,4 @@
 libaudiodec_la_LDFLAGS = @AC3_LIBS@ @AC3_LDADD@ -module
 CXXFLAGS = @CXXFLAGS@ @AC3_CFLAGS@
 
-LIBS=
+LIBS=-L../../lib/.libs -laviplay -lstdc++
diff -urN avifile-0.6.org/plugins/libmp3lame_audioenc/Makefile.am avifile-0.6/plugins/libmp3lame_audioenc/Makefile.am
--- avifile-0.6.org/plugins/libmp3lame_audioenc/Makefile.am	Fri Mar 23 17:27:10 2001
+++ avifile-0.6/plugins/libmp3lame_audioenc/Makefile.am	Fri Mar 23 17:27:20 2001
@@ -4,4 +4,4 @@
 libmp3lame_audioenc_la_SOURCES = audioencoder.cpp mp3encoder.cpp mp3encoder.h fillplugins.h
 libmp3lame_audioenc_la_LDFLAGS = -module
 
-LIBS=
+LIBS=-L../../lib/.libs -laviplay -lstdc++
diff -urN avifile-0.6.org/plugins/libmpeg_audiodec/Makefile.am avifile-0.6/plugins/libmpeg_audiodec/Makefile.am
--- avifile-0.6.org/plugins/libmpeg_audiodec/Makefile.am	Fri Mar 23 17:27:10 2001
+++ avifile-0.6/plugins/libmpeg_audiodec/Makefile.am	Fri Mar 23 17:27:20 2001
@@ -9,4 +9,4 @@
 
 CPPFLAGS = @CPPFLAGS@
 
-LIBS=
+LIBS=-L../../lib/.libs -laviplay -lstdc++
diff -urN avifile-0.6.org/plugins/libwin32/Makefile.am avifile-0.6/plugins/libwin32/Makefile.am
--- avifile-0.6.org/plugins/libwin32/Makefile.am	Fri Mar 23 17:27:10 2001
+++ avifile-0.6/plugins/libwin32/Makefile.am	Fri Mar 23 17:28:27 2001
@@ -13,4 +13,4 @@
 
 CPPFLAGS = @CPPFLAGS@ -Iloader -Iloader/wine -Ivideocodec -Iaudiodec -IDirectShow
 
-LIBS = @AVILIBDEPLIB@
+LIBS = @AVILIBDEPLIB@  -L../../lib/.libs -laviplay -lstdc++
