diff -urN avifile-0.6.0.org/Makefile.am avifile-0.6.0/Makefile.am
--- avifile-0.6.0.org/Makefile.am	Fri Feb 23 14:42:33 2001
+++ avifile-0.6.0/Makefile.am	Fri Feb 23 16:36:30 2001
@@ -1,8 +1,8 @@
 BASE_SUBDIRS = 
 if AM_USE_QT
-EXTRA_SUBDIRS = include plugins lib player samples bin doc
+EXTRA_SUBDIRS = include lib plugins player samples bin doc
 else
-EXTRA_SUBDIRS = include plugins lib samples bin doc
+EXTRA_SUBDIRS = include lib plugins samples bin doc
 endif
 SUBDIRS=$(BASE_SUBDIRS) $(EXTRA_SUBDIRS)
 
diff -urN avifile-0.6.0.org/plugins/libaudiodec/Makefile.am avifile-0.6.0/plugins/libaudiodec/Makefile.am
--- avifile-0.6.0.org/plugins/libaudiodec/Makefile.am	Fri Feb 23 14:42:33 2001
+++ avifile-0.6.0/plugins/libaudiodec/Makefile.am	Fri Feb 23 17:06:39 2001
@@ -7,4 +7,4 @@
 libaudiodec_la_LDFLAGS=@AC3_LIBS@ @AC3_LDADD@ -module
 CXXFLAGS=-I../../include @AC3_CFLAGS@
 
-LIBS=
+LIBS=-L../../lib/.libs -laviplay -lstdc++
diff -urN avifile-0.6.0.org/plugins/libmp3lame_audioenc/Makefile.am avifile-0.6.0/plugins/libmp3lame_audioenc/Makefile.am
--- avifile-0.6.0.org/plugins/libmp3lame_audioenc/Makefile.am	Fri Feb 23 14:42:33 2001
+++ avifile-0.6.0/plugins/libmp3lame_audioenc/Makefile.am	Fri Feb 23 17:06:30 2001
@@ -5,4 +5,4 @@
 libmp3lame_audioenc_la_LDFLAGS=-module
 CXXFLAGS=-I../../include
 
-LIBS=
+LIBS=-L../../lib/.libs -laviplay -lstdc++
diff -urN avifile-0.6.0.org/plugins/libmpeg_audiodec/Makefile.am avifile-0.6.0/plugins/libmpeg_audiodec/Makefile.am
--- avifile-0.6.0.org/plugins/libmpeg_audiodec/Makefile.am	Fri Feb 23 14:42:33 2001
+++ avifile-0.6.0/plugins/libmpeg_audiodec/Makefile.am	Fri Feb 23 17:06:23 2001
@@ -6,4 +6,4 @@
 noinst_HEADERS=mpegsound.h mpegsound_locals.h fillplugins.h
 CXXFLAGS=-I../../include -fno-exceptions
 
-LIBS=
+LIBS=-L../../lib/.libs -laviplay -lstdc++
diff -urN avifile-0.6.0.org/plugins/libwin32/Makefile.am avifile-0.6.0/plugins/libwin32/Makefile.am
--- avifile-0.6.0.org/plugins/libwin32/Makefile.am	Fri Feb 23 14:42:33 2001
+++ avifile-0.6.0/plugins/libwin32/Makefile.am	Fri Feb 23 17:06:13 2001
@@ -6,4 +6,4 @@
   audiodec/libaudiodec.la DirectShow/libds_filter.la
 libwin32_la_LDFLAGS = -module -version-info 0:0:0
 CXXFLAGS=-I- -I. -I../../include -Iloader -Iloader/wine -Ivideocodec -Iaudiodec -IDirectShow
-LIBS= @AVILIBDEPLIB@
+LIBS= -L../../lib/.libs -laviplay @AVILIBDEPLIB@ -lstdc++
