--- avifile-0.7-0.7.45/ffmpeg/libavcodec/ppc/Makefile.am.orig	2005-08-28 22:58:50.000000000 +0200
+++ avifile-0.7-0.7.45/ffmpeg/libavcodec/ppc/Makefile.am	2006-03-09 00:01:32.960222500 +0100
@@ -23,6 +23,6 @@
 libavcodecppc_la_SOURCES = $(PPC_SOURCES)
 endif
 
-AM_CPPFLAGS = $(LTNOPIC) -DHAVE_AV_CONFIG_H -I$(srcdir)/../../libavutil -I$(srcdir)/../..
+AM_CPPFLAGS = $(LTNOPIC) -DHAVE_AV_CONFIG_H -DHAVE_ALTIVEC_H -DHAVE_ALTIVEC -maltivec -mabi=altivec -I$(srcdir)/../../libavutil -I$(srcdir)/../..
 
 MAINTAINERCLEANFILES = Makefile.in
