--- avifile-0.6/plugins/libffmpeg/libavcodec/Makefile.am.wiget	Mon Jul 16 16:02:37 2001
+++ avifile-0.6/plugins/libffmpeg/libavcodec/Makefile.am	Mon Jul 16 16:02:37 2001
@@ -1,14 +1,15 @@
 noinst_LTLIBRARIES = libavcodec.la
 
-I386_SOURCE_FILES = 386/fdct_mmx.s i386/fdctdata.s i386/sad_mmx.s
+I386_SOURCE_FILES_C = i386/fdctdata.c
+I386_SOURCE_FILES_S = i386/fdct_mmx.s i386/sad_mmx.s
 
 libavcodec_la_SOURCES = common.c utils.c mpegvideo.c h263.c jrevdct.c \
       jfdctfst.c mpegaudio.c ac3enc.c mjpegenc.c resample.c dsputil.c \
       motion_est.c imgconvert.c imgresample.c msmpeg4.c \
-      mpeg12.c h263dec.c mpegaudiodec.c $(I386_SOURCE_FILES)
+      mpeg12.c h263dec.c mpegaudiodec.c 
 
 libavcodec_la_LDFLAGS = -module
-
+libavcodec_la_LIBADD = $(patsubst %.c, %.o, $(I386_SOURCE_FILES_C)) $(patsubst %.s, %.o, $(I386_SOURCE_FILES_S))
 LIBS =
 
 # i386 specific stuff
@@ -16,5 +17,5 @@
 # SSE can also be added
 #CFLAGS += -DCONFIG_SSE
 
-#%.o: %.s
-#	nasm -f elf -o $@ $<
+%.o: %.s
+	nasm -f elf -o $@ $<
--- avifile-0.6/acinclude.m4.wiget	Mon Jul 16 16:03:09 2001
+++ avifile-0.6/acinclude.m4	Mon Jul 16 16:04:20 2001
@@ -415,8 +415,8 @@
 AC_CACHE_VAL(ac_cv_have_x,
 [# One or both of the vars are not set, and there is no cached value.
 ac_x_includes=NO ac_x_libraries=NO
-AC_PATH_X_DIRECT
-AC_PATH_X_XMKMF
+_AC_PATH_X_DIRECT
+_AC_PATH_X_XMKMF
 if test "$ac_x_includes" = NO || test "$ac_x_libraries" = NO; then
   AC_MSG_ERROR([Can't find X includes. Please check your installation and add the correct paths!])
 else
@@ -1662,52 +1662,6 @@
 AC_SUBST($1)dnl
 ])
 
-
-# Check whether LC_MESSAGES is available in <locale.h>.
-# Ulrich Drepper <drepper@cygnus.com>, 1995.
- 
-# serial 1
- 
-AC_DEFUN(AM_LC_MESSAGES,
-  [if test $ac_cv_header_locale_h = yes; then
-    AC_CACHE_CHECK([for LC_MESSAGES], am_cv_val_LC_MESSAGES,
-      [AC_TRY_LINK([#include <locale.h>], [return LC_MESSAGES],
-       am_cv_val_LC_MESSAGES=yes, am_cv_val_LC_MESSAGES=no)])
-    if test $am_cv_val_LC_MESSAGES = yes; then
-      AC_DEFINE(HAVE_LC_MESSAGES)
-    fi
-  fi])
- 
-dnl From Jim Meyering.
-dnl FIXME: migrate into libit.
-
-AC_DEFUN(AM_FUNC_OBSTACK,
-[AC_CACHE_CHECK([for obstacks], am_cv_func_obstack,
- [AC_TRY_LINK([#include "obstack.h"],
-	      [struct obstack *mem;obstack_free(mem,(char *) 0)],
-	      am_cv_func_obstack=yes,
-	      am_cv_func_obstack=no)])
- if test $am_cv_func_obstack = yes; then
-   AC_DEFINE(HAVE_OBSTACK)
- else
-   LIBOBJS="$LIBOBJS obstack.o"
- fi
-])
-
-dnl From Jim Meyering.  Use this if you use the GNU error.[ch].
-dnl FIXME: Migrate into libit
-
-AC_DEFUN(AM_FUNC_ERROR_AT_LINE,
-[AC_CACHE_CHECK([for error_at_line], am_cv_lib_error_at_line,
- [AC_TRY_LINK([],[error_at_line(0, 0, "", 0, "");],
-              am_cv_lib_error_at_line=yes,
-	      am_cv_lib_error_at_line=no)])
- if test $am_cv_lib_error_at_line = no; then
-   LIBOBJS="$LIBOBJS error.o"
- fi
- AC_SUBST(LIBOBJS)dnl
-])
-
 # Macro to add for using GNU gettext.
 # Ulrich Drepper <drepper@cygnus.com>, 1995.
 
