diff -urN aa/avifile-0.7-0.7.38/include/aviplay.h avifile-0.7-0.7.38/include/aviplay.h
--- aa/avifile-0.7-0.7.38/include/aviplay.h	2003-05-25 00:51:30.000000000 +0200
+++ avifile-0.7-0.7.38/include/aviplay.h	2004-07-02 15:03:57.497990112 +0200
@@ -12,8 +12,8 @@
 #include "image.h"
 #include "avm_args.h"
 
-typedef void (*KILLHANDLER)(int, void* p = 0);
-typedef int (*AUDIOFUNC)(void* srcdata, unsigned int size, void* p = 0);
+typedef void (*KILLHANDLER)(int, void* p);
+typedef int (*AUDIOFUNC)(void* srcdata, unsigned int size, void* p);
 
 AVM_BEGIN_NAMESPACE;
 
diff -urN aa/avifile-0.7-0.7.38/include/avm_map.h avifile-0.7-0.7.38/include/avm_map.h
--- aa/avifile-0.7-0.7.38/include/avm_map.h	2003-03-27 02:01:22.000000000 +0100
+++ avifile-0.7-0.7.38/include/avm_map.h	2004-07-02 15:15:04.427601400 +0200
@@ -198,7 +198,8 @@
 {
     // cast to the needed type - used to prevent internal compiler error
     // for old egcc
-    avm_map<Key, Value, Compare, Equal>::_Tnode* node = (avm_map<Key, Value, Compare, Equal>::_Tnode*) n;
+    //avm_map<Key, Value, Compare, Equal>::_Tnode* node = (avm_map<Key, Value, Compare, Equal>::_Tnode*) n;
+    _Tnode* node = (_Tnode*) n;
 
     if(node->entry)
     {
diff -urN aa/avifile-0.7-0.7.38/include/avm_stl.h avifile-0.7-0.7.38/include/avm_stl.h
--- aa/avifile-0.7-0.7.38/include/avm_stl.h	2003-06-08 17:57:05.000000000 +0200
+++ avifile-0.7-0.7.38/include/avm_stl.h	2004-07-02 15:12:53.351527992 +0200
@@ -74,6 +74,7 @@
 	m_Type = new Type[m_uiCapacity];
     }
     ~qring() { delete[] m_Type; }
+    qring<Type>(const qring<Type>& t) {}
     qring<Type>& operator=(const qring<Type>& t)
     {
 	if (this != &t)
@@ -149,7 +150,6 @@
 
     // disabled for now
     qring<Type>() {}
-    qring<Type>(const qring<Type>& t) {}
 
     uint_t bpos() const { return (m_uiPos > 0) ? m_uiPos - 1 : m_uiCapacity - 1; }
     uint_t fpos() const { return (m_uiSize > m_uiPos) ? m_uiPos + m_uiCapacity - m_uiSize: m_uiPos - m_uiSize; }
diff -urN aa/avifile-0.7-0.7.38/lib/aviread/AsfNetworkInputStream.cpp avifile-0.7-0.7.38/lib/aviread/AsfNetworkInputStream.cpp
--- aa/avifile-0.7-0.7.38/lib/aviread/AsfNetworkInputStream.cpp	2003-05-27 17:26:31.000000000 +0200
+++ avifile-0.7-0.7.38/lib/aviread/AsfNetworkInputStream.cpp	2004-07-02 15:08:12.642202296 +0200
@@ -793,22 +793,24 @@
     return -1;
 }
 
-int AsfNetworkInputStream::write(const void* buffer, uint_t size)
+int AsfNetworkInputStream::write(const void* b, uint_t size)
 {
+    const char* buffer = (const char*) b;
     int wsize = size;
     while (wsize > 0)
     {
 	int i = ::write(m_iSocket, buffer, wsize);
 	if (i <= 0)
             return i;
-	(const char*)buffer += i;
+	buffer += i;
         wsize -= i;
     }
     return size;
 }
 
-int AsfNetworkInputStream::dwrite(const void* buffer, uint_t size)
+int AsfNetworkInputStream::dwrite(const void* b, uint_t size)
 {
+    const char* buffer = (const char*) b;
     if (m_lfd < 0)
     {
 	if (m_lfd == -12345)
@@ -843,7 +845,7 @@
 	int i = ::write(m_lfd, buffer, size);
 	if (i < 0)
             return i;
-	(const char*)buffer += i;
+	buffer += i;
 	size -= i;
     }
     fsync(m_lfd);
diff -urN aa/avifile-0.7-0.7.38/lib/common/image.cpp avifile-0.7-0.7.38/lib/common/image.cpp
--- aa/avifile-0.7-0.7.38/lib/common/image.cpp	2003-05-25 01:00:58.000000000 +0200
+++ avifile-0.7-0.7.38/lib/common/image.cpp	2004-07-02 15:16:53.695990080 +0200
@@ -954,7 +954,7 @@
     fillMembers();
     if (!copy)
     {
-	(const uint8_t*) m_pPlane[0] = data;
+	m_pPlane[0] = (uint8_t*) data;
     }
     else
     {
diff -urN aa/avifile-0.7-0.7.38/plugins/libvorbis/libvorbis.cpp avifile-0.7-0.7.38/plugins/libvorbis/libvorbis.cpp
--- aa/avifile-0.7-0.7.38/plugins/libvorbis/libvorbis.cpp	2003-05-20 16:33:13.000000000 +0200
+++ avifile-0.7-0.7.38/plugins/libvorbis/libvorbis.cpp	2004-07-02 15:19:50.622093224 +0200
@@ -96,7 +96,7 @@
 	op.packet = NULL;
 	op.b_o_s  = 1; /* beginning of stream for first packet */
 	op.bytes  = hdrsizes[0];
-	(const void*) op.packet = vorbishdr;
+	op.packet = (unsigned char*) vorbishdr;
 	vorbishdr += op.bytes;
 	if (vorbis_synthesis_headerin(&vi, &vc, &op) < 0)
 	{
@@ -106,7 +106,7 @@
 
 	op.b_o_s  = 0;
 	op.bytes  = hdrsizes[1];
-	(const void*) op.packet = vorbishdr;
+	op.packet = (unsigned char*) vorbishdr;
 	vorbishdr += op.bytes;
 	if (vorbis_synthesis_headerin(&vi, &vc, &op) < 0)
 	{
@@ -115,7 +115,7 @@
 	}
 
 	op.bytes  = hdrsizes[2];
-	(const void*) op.packet = vorbishdr;
+	op.packet = (unsigned char*) vorbishdr;
 	vorbishdr += op.bytes;
 	if (vorbis_synthesis_headerin(&vi, &vc, &op) < 0)
 	{
