diff -urN avifile-0.50.org/samples/benchmark/Makefile.in avifile-0.50/samples/benchmark/Makefile.in
--- avifile-0.50.org/samples/benchmark/Makefile.in	Mon Oct 30 00:43:43 2000
+++ avifile-0.50/samples/benchmark/Makefile.in	Mon Oct 30 00:48:17 2000
@@ -6,7 +6,7 @@
     
 
 LIBS=@LIBS@ -lX11 -lXext @QT_NAME@ -laviplay @DGA_LDADD@ @SDL_LIBS@
-LIB_DIRS=@QT_LDFLAGS@ -L/usr/X11R6/lib -L../../lib
+LIB_DIRS=-L../../lib @QT_LDFLAGS@ -L/usr/X11R6/lib -L../../lib
 
 MOC=@MOC@
 CDOPT=@CD_OPT@
diff -urN avifile-0.50.org/samples/extractor/Makefile.in avifile-0.50/samples/extractor/Makefile.in
--- avifile-0.50.org/samples/extractor/Makefile.in	Mon Oct 30 00:43:43 2000
+++ avifile-0.50/samples/extractor/Makefile.in	Mon Oct 30 00:48:29 2000
@@ -1,5 +1,5 @@
 CPPFLAGS=-g -I../../include
-LIBS=@X_LDFLAGS@ @LIBS@ -laviplay -L../../lib
+LIBS= -L../../lib @X_LDFLAGS@ @LIBS@ -laviplay -L../../lib
 
 OBJS=extractor.o
 
