diff -Nur avifile0.7-0.7.7.orig/include/fourcc.h avifile0.7-0.7.7/include/fourcc.h
--- avifile0.7-0.7.7.orig/include/fourcc.h	Fri May 10 14:54:32 2002
+++ avifile0.7-0.7.7/include/fourcc.h	Wed Jun 12 15:03:51 2002
@@ -106,7 +106,8 @@
 #define RIFFINFO_ASV2	mmioFOURCC('A', 'S', 'V', '2')
 
 /* let's give some FCC to Xvid */
-#define RIFFINFO_XVID   mmioFOURCC('X', 'v', 'i', 'D')
+#define RIFFINFO_XviD   mmioFOURCC('X', 'v', 'i', 'D')
+#define RIFFINFO_XVID   mmioFOURCC('X', 'V', 'I', 'D')
 
 /* Microsoft video */
 #define RIFFINFO_CRAM	mmioFOURCC('C', 'R', 'A', 'M')
@@ -239,6 +240,7 @@
 #define fccMJPG		RIFFINFO_MJPG
 #define fccmjpg		RIFFINFO_mjpg
 #define fccHFYU		RIFFINFO_HFYU
+#define fccXviD		RIFFINFO_XviD
 #define fccXVID		RIFFINFO_XVID
 #define fccCVID		RIFFINFO_CVID
 #define fcccvid		RIFFINFO_cvid
diff -Nur avifile0.7-0.7.7.orig/plugins/libwin32/fillplugins.h avifile0.7-0.7.7/plugins/libwin32/fillplugins.h
--- avifile0.7-0.7.7.orig/plugins/libwin32/fillplugins.h	Sat May 11 17:52:57 2002
+++ avifile0.7-0.7.7/plugins/libwin32/fillplugins.h	Wed Jun 12 15:03:51 2002
@@ -401,6 +401,7 @@
         "http://www.geocities.com/avihpit/xvid/post.html";
     static const fourcc_t xvid_codecs[] =
     {
+	fccXviD,
 	fccXVID,
 	fccDIVX, fccdivx, 0
     };
diff -Nur avifile0.7-0.7.7.orig/plugins/libwin32/libwin32.cpp avifile0.7-0.7.7/plugins/libwin32/libwin32.cpp
--- avifile0.7-0.7.7.orig/plugins/libwin32/libwin32.cpp	Wed May  8 15:28:52 2002
+++ avifile0.7-0.7.7/plugins/libwin32/libwin32.cpp	Wed Jun 12 15:03:51 2002
@@ -284,6 +284,7 @@
 	    || strcmp(attribute, "strSettings") == 0)
 	    return GetRegValue(vp31name, attribute, 0, &value, 0);
         break;
+    case fccXviD:
     case fccXVID:
 	return GetRegValue(xvidname, attribute, 0, &value, 0);
     case fccMJPG:
@@ -371,6 +372,7 @@
 	    || strcmp(attribute, "strSettings") == 0)
 	    return SetRegValue(vp31name, attribute, value, 0);
         break;
+    case fccXviD:
     case fccXVID:
 	return SetRegValue(xvidname, attribute, value, 0);
     case fccMJPG:
diff -Nur avifile0.7-0.7.7.orig/plugins/libwin32/videocodec/VideoDecoder.cpp avifile0.7-0.7.7/plugins/libwin32/videocodec/VideoDecoder.cpp
--- avifile0.7-0.7.7.orig/plugins/libwin32/videocodec/VideoDecoder.cpp	Mon Apr  8 09:47:51 2002
+++ avifile0.7-0.7.7/plugins/libwin32/videocodec/VideoDecoder.cpp	Wed Jun 12 15:03:51 2002
@@ -402,6 +402,7 @@
 
     switch (m_Info.fourcc)
     {
+    case fccXviD:
     case fccXVID:
 	if (m_bitrick.biCompression == 0 || m_bitrick.biCompression == 3)
 	    break;
diff -Nur avifile0.7-0.7.7.orig/plugins/libxvid/fillplugins.h avifile0.7-0.7.7/plugins/libxvid/fillplugins.h
--- avifile0.7-0.7.7.orig/plugins/libxvid/fillplugins.h	Wed May  1 17:34:06 2002
+++ avifile0.7-0.7.7/plugins/libxvid/fillplugins.h	Wed Jun 12 15:05:02 2002
@@ -44,6 +44,7 @@
     static char xvid_about[] = "XviD MPEG-4 Video codec";
     static const fourcc_t xvid_codecs[] =
     {
+        fccXviD, 
         fccXVID, fccDIVX, fccdivx, 0
     };
 
