diff -Nur avifile-0.6.orig/plugins/libac3pass/Makefile.am avifile-0.6/plugins/libac3pass/Makefile.am
--- avifile-0.6.orig/plugins/libac3pass/Makefile.am	Tue Aug 21 18:08:30 2001
+++ avifile-0.6/plugins/libac3pass/Makefile.am	Tue Oct  2 21:13:04 2001
@@ -3,7 +3,7 @@
 
 libac3pass_la_SOURCES = ac3-iec958.c libac3pass.cpp
 
-libac3pass_la_LIBADD =
+libac3pass_la_LIBADD = ../../lib/libaviplay.la -lstdc++
 libac3pass_la_LDFLAGS = -module -release $(LT_RELEASE) -version-info $(LT_VERSION)
 
 CLEANFILES = play-ac3
diff -Nur avifile-0.6.orig/plugins/libaudiodec/Makefile.am avifile-0.6/plugins/libaudiodec/Makefile.am
--- avifile-0.6.orig/plugins/libaudiodec/Makefile.am	Thu Aug  9 18:54:04 2001
+++ avifile-0.6/plugins/libaudiodec/Makefile.am	Tue Oct  2 21:13:16 2001
@@ -3,7 +3,7 @@
 	gsm61dec.cpp pcm.cpp aulaw.cpp xa_gsm.c
 noinst_HEADERS = ac3.h adpcm.h adpcm_impl.h gsm61dec.h pcm.h xa_gsm.h \
 	xa_gsm_state.h aulaw.h fillplugins.h
-libaudiodec_la_LIBADD = @AC3_LIBADD@
+libaudiodec_la_LIBADD = @AC3_LIBADD@ ../../lib/libaviplay.la -lstdc++
 libaudiodec_la_LDFLAGS = @AC3_LIBS@ @AC3_LDADD@ -module -release $(LT_RELEASE) -version-info $(LT_VERSION)
 
 CXXFLAGS = @CXXFLAGS@ @AC3_CFLAGS@
diff -Nur avifile-0.6.orig/plugins/libdivx4/Makefile.am avifile-0.6/plugins/libdivx4/Makefile.am
--- avifile-0.6.orig/plugins/libdivx4/Makefile.am	Tue Aug 21 19:34:22 2001
+++ avifile-0.6/plugins/libdivx4/Makefile.am	Tue Oct  2 21:13:41 2001
@@ -1,6 +1,7 @@
 pkglib_LTLIBRARIES=libdivx4.la
 libdivx4_la_SOURCES=divx4.cpp
 libdivx4_la_LDFLAGS=-L/usr/local/lib -module -ldivxdecore -ldivxencore -release $(LT_RELEASE) -version-info $(LT_VERSION)
+libdivx4_la_LIBADD=../../lib/libaviplay.la -lstdc++
 CXXFLAGS=-I$(top_srcdir)/include -g -DLINUX
 noinst_HEADERS=fillplugins.h
 noinst_PROGRAMS=test 
diff -Nur avifile-0.6.orig/plugins/libffmpeg/Makefile.am avifile-0.6/plugins/libffmpeg/Makefile.am
--- avifile-0.6.orig/plugins/libffmpeg/Makefile.am	Thu Aug  9 18:54:05 2001
+++ avifile-0.6/plugins/libffmpeg/Makefile.am	Tue Oct  2 21:14:02 2001
@@ -2,7 +2,7 @@
 pkglib_LTLIBRARIES = libffmpeg.la
 noinst_HEADERS = FFAudioDecoder.h FFVideoDecoder.h fillplugins.h bswap.h
 libffmpeg_la_SOURCES = libffmpeg.cpp FFAudioDecoder.cpp FFVideoDecoder.cpp 
-libffmpeg_la_LIBADD = libavcodec/libavcodec.la
+libffmpeg_la_LIBADD = libavcodec/libavcodec.la ../../lib/libaviplay.la -lstdc++
 libffmpeg_la_LDFLAGS = -module -release $(LT_RELEASE) -version-info $(LT_VERSION)
 
 #
diff -Nur avifile-0.6.orig/plugins/libmp3lame_audioenc/Makefile.am avifile-0.6/plugins/libmp3lame_audioenc/Makefile.am
--- avifile-0.6.orig/plugins/libmp3lame_audioenc/Makefile.am	Thu Aug  9 18:54:05 2001
+++ avifile-0.6/plugins/libmp3lame_audioenc/Makefile.am	Tue Oct  2 21:14:16 2001
@@ -2,5 +2,5 @@
 pkglib_LTLIBRARIES = libmp3lame_audioenc.la
 noinst_HEADERS = mp3encoder.h fillplugins.h
 libmp3lame_audioenc_la_SOURCES = audioencoder.cpp mp3encoder.cpp 
-libmp3lame_audioenc_la_LIBADD = lame3.70/libmp3lame.la
+libmp3lame_audioenc_la_LIBADD = lame3.70/libmp3lame.la ../../lib/libaviplay.la -lstdc++
 libmp3lame_audioenc_la_LDFLAGS = -module -release $(LT_RELEASE) -version-info $(LT_VERSION)
diff -Nur avifile-0.6.orig/plugins/libmp3lamebin_audioenc/Makefile.am avifile-0.6/plugins/libmp3lamebin_audioenc/Makefile.am
--- avifile-0.6.orig/plugins/libmp3lamebin_audioenc/Makefile.am	Thu Aug  9 18:54:05 2001
+++ avifile-0.6/plugins/libmp3lamebin_audioenc/Makefile.am	Tue Oct  2 21:14:40 2001
@@ -2,7 +2,7 @@
 noinst_HEADERS = fillplugins.h lame.h lameencoder.h
 libmp3lamebin_audioenc_la_SOURCES = audioencoder.cpp lameencoder.cpp
 libmp3lamebin_audioenc_la_LDFLAGS = -module -release $(LT_RELEASE) -version-info $(LT_VERSION)
-
+libmp3lamebin_audioenc_la_LIBADD = ../../lib/libaviplay.la -lstdc++
 #
 # only lame.h header from whole LAME project is here
 #
diff -Nur avifile-0.6.orig/plugins/libmpeg_audiodec/Makefile.am avifile-0.6/plugins/libmpeg_audiodec/Makefile.am
--- avifile-0.6.orig/plugins/libmpeg_audiodec/Makefile.am	Thu Aug  9 18:54:05 2001
+++ avifile-0.6/plugins/libmpeg_audiodec/Makefile.am	Tue Oct  2 21:15:19 2001
@@ -3,10 +3,10 @@
 	huffmantable.cpp mpeglayer1.cpp mpeglayer2.cpp mpeglayer3.cpp \
 	mpegtable.cpp mpegtoraw.cpp
 libmpeg_audiodec_la_LDFLAGS = -module -release $(LT_RELEASE) -version-info $(LT_VERSION)
+libmpeg_audiodec_la_LIBADD = ../../lib/libaviplay.la -lstdc++
 noinst_HEADERS = mpegsound.h mpegsound_locals.h fillplugins.h
 
 # currently it has to be compiled with optimalizations (inline has to work)
-OPTFLG = -O2 -ffast-math -fomit-frame-pointer -funroll-loops
 #-D__NO_STRING_INLINES -DDEBUG
 # -finline-functions 
-CXXFLAGS = $(OPTFLG) @CXXFLAGS@ 
+CXXFLAGS = @CXXFLAGS@ 
diff -Nur avifile-0.6.orig/plugins/libvorbis/Makefile.am avifile-0.6/plugins/libvorbis/Makefile.am
--- avifile-0.6.orig/plugins/libvorbis/Makefile.am	Tue Sep  4 22:28:57 2001
+++ avifile-0.6/plugins/libvorbis/Makefile.am	Tue Oct  2 21:15:32 2001
@@ -3,6 +3,6 @@
 
 libvorbis_audio_la_SOURCES = libvorbis.cpp
 
-libvorbis_audio_la_LIBADD = -lvorbisfile -lvorbis
+libvorbis_audio_la_LIBADD = -lvorbisfile -lvorbis ../../lib/libaviplay.la -lstdc++
 ##libvorbis_audio_la_LIBADD = /usr/local/lib/libvorbisfile.a /usr/local/lib/libvorbis.a -logg
 libvorbis_audio_la_LDFLAGS = -module -release $(LT_RELEASE) -version-info $(LT_VERSION)
diff -Nur avifile-0.6.orig/plugins/libwin32/Makefile.am avifile-0.6/plugins/libwin32/Makefile.am
--- avifile-0.6.orig/plugins/libwin32/Makefile.am	Thu Aug  9 18:54:05 2001
+++ avifile-0.6/plugins/libwin32/Makefile.am	Tue Oct  2 21:15:45 2001
@@ -8,7 +8,7 @@
   DirectShow/libds_filter.la \
   audiodec/libaudiodec.la \
   loader/libloader.la \
-  videocodec/libvideocodec.la
+  videocodec/libvideocodec.la ../../lib/libaviplay.la -lstdc++
 
 libwin32_la_LDFLAGS = -module -release $(LT_RELEASE) -version-info $(LT_VERSION)
 
