--- avifile/configure.in	Tue Mar 18 13:17:08 2003
+++ avifile/configure.in~	Fri Apr  4 14:17:38 2003
@@ -168,7 +168,7 @@
 AC_CHECK_LIB(dl, dlopen)
 AC_CHECK_LIB(posix4, nanosleep)
 AC_CHECK_LIB(socket, socket)
-AC_CHECK_LIB(nsl, gethostbyname)
+AC_CHECK_FUNC(gethostbyname, [], AC_CHECK_LIB(nsl, gethostbyname))
 AC_CHECK_LIB(kstat, kstat_open)
 AC_CHECK_LIB(c, lseek64, AC_DEFINE(HAVE_LSEEK64, 1, [Define if you have 'lseek64' function.]))
 dnl AC_SUBST(LIBSOCKET)
