--- avifile-0.7-0.7.38/ffmpeg/libavcodec/ppc/Makefile.am.orig	2003-05-25 23:11:57.000000000 +0200
+++ avifile-0.7-0.7.38/ffmpeg/libavcodec/ppc/Makefile.am	2003-11-14 01:06:03.904622008 +0100
@@ -20,6 +20,6 @@
 
 libavcodecppc_la_SOURCES = $(PPC_SRC)
 
-AM_CPPFLAGS = $(LTNOPIC) -DHAVE_AV_CONFIG_H -I$(srcdir)/../..
+AM_CPPFLAGS = $(LTNOPIC) -DHAVE_AV_CONFIG_H -DHAVE_ALTIVEC_H -DHAVE_ALTIVEC -maltivec -mabi=altivec -I$(srcdir)/../..
 
 MAINTAINERCLEANFILES = Makefile.in
