diff -urN avifile-0.50.org/Makefile.in avifile-0.50/Makefile.in
--- avifile-0.50.org/Makefile.in	Mon Oct 30 11:45:32 2000
+++ avifile-0.50/Makefile.in	Mon Oct 30 11:46:10 2000
@@ -49,49 +49,48 @@
 	-find . -name Makefile -exec rm \{\} \;
 
 stable-install:
-	$(INSTALL_PROGRAM) -d $(datadir)/avifile
-	$(INSTALL_PROGRAM) -d $(libdir)
-	$(INSTALL_PROGRAM) -d $(bindir)
-	$(INSTALL_PROGRAM) ./lib/libaviplay.so $(libdir)
+	$(INSTALL_PROGRAM) -d $(DESTDIR)$(datadir)/avifile
+	$(INSTALL_PROGRAM) -d $(DESTDIR)$(libdir)
+	$(INSTALL_PROGRAM) -d $(DESTDIR)$(bindir)
+	$(INSTALL_PROGRAM) ./lib/libaviplay.so $(DESTDIR)$(libdir)
 ifeq (@USE_QT@,1)
-	$(INSTALL_PROGRAM) ./bin/aviplay $(bindir)
-	$(INSTALL_PROGRAM) ./bin/benchmark $(bindir)
-	$(INSTALL_PROGRAM) ./bin/*.ppm $(datadir)/avifile
-	$(INSTALL_PROGRAM) ./bin/*.png $(datadir)/avifile
+	$(INSTALL_PROGRAM) ./bin/aviplay $(DESTDIR)$(bindir)
+	$(INSTALL_PROGRAM) ./bin/benchmark $(DESTDIR)$(bindir)
+	$(INSTALL_PROGRAM) ./bin/*.ppm $(DESTDIR)$(datadir)/avifile
+	$(INSTALL_PROGRAM) ./bin/*.png $(DESTDIR)$(datadir)/avifile
 ifeq (@USE_VIDCAP@,1)
-	$(INSTALL_PROGRAM) ./bin/qtvidcap $(bindir)
-	$(INSTALL_PROGRAM) -m4755 ./bin/kv4lsetup $(bindir)
+	$(INSTALL_PROGRAM) ./bin/qtvidcap $(DESTDIR)$(bindir)
+	$(INSTALL_PROGRAM) -m4755 ./bin/kv4lsetup $(DESTDIR)$(bindir)
 endif	
 endif
-	$(INSTALL_PROGRAM) -m0755 ./avifile-config $(bindir)
+	$(INSTALL_PROGRAM) -m0755 ./avifile-config $(DESTDIR)$(bindir)
 
 install:
-	-$(INSTALL_PROGRAM) -d $(datadir)
-	-$(INSTALL_PROGRAM) -d $(includedir)/avifile/wine
-	-$(INSTALL_PROGRAM) -d $(datadir)/avifile
-	-$(INSTALL_PROGRAM) -d $(libdir)
-	-$(INSTALL_PROGRAM) -d $(bindir)
-	-$(INSTALL_PROGRAM) ./lib/libaviplay.so $(libdir)
-	-$(INSTALL_PROGRAM) ./include/*.h $(includedir)/avifile
-	-$(INSTALL_PROGRAM) ./include/wine/*.h $(includedir)/avifile/wine
+	-$(INSTALL_PROGRAM) -d $(DESTDIR)$(datadir)
+	-$(INSTALL_PROGRAM) -d $(DESTDIR)$(includedir)/avifile/wine
+	-$(INSTALL_PROGRAM) -d $(DESTDIR)$(datadir)/avifile
+	-$(INSTALL_PROGRAM) -d $(DESTDIR)$(libdir)
+	-$(INSTALL_PROGRAM) -d $(DESTDIR)$(bindir)
+	-$(INSTALL_PROGRAM) ./lib/libaviplay.so $(DESTDIR)$(libdir)
+	-$(INSTALL_PROGRAM) ./include/*.h $(DESTDIR)$(includedir)/avifile
+	-$(INSTALL_PROGRAM) ./include/wine/*.h $(DESTDIR)$(includedir)/avifile/wine
 ifeq (@USE_QT@,1)	
-	-$(INSTALL_PROGRAM) ./bin/aviplay $(bindir)
-	-$(INSTALL_PROGRAM) ./bin/benchmark $(bindir)
-	-$(INSTALL_PROGRAM) ./bin/*.ppm $(datadir)/avifile
-	-$(INSTALL_PROGRAM) ./bin/*.png $(datadir)/avifile
+	-$(INSTALL_PROGRAM) ./bin/aviplay $(DESTDIR)$(bindir)
+	-$(INSTALL_PROGRAM) ./bin/benchmark $(DESTDIR)$(bindir)
+	-$(INSTALL_PROGRAM) ./bin/*.ppm $(DESTDIR)$(datadir)/avifile
+	-$(INSTALL_PROGRAM) ./bin/*.png $(DESTDIR)$(datadir)/avifile
 ifeq (@USE_VIDCAP@,1)
-	-$(INSTALL_PROGRAM) ./bin/qtvidcap $(bindir)
-	-$(INSTALL_PROGRAM) -m4755 ./bin/kv4lsetup $(bindir)
+	-$(INSTALL_PROGRAM) ./bin/qtvidcap $(DESTDIR)$(bindir)
+	-$(INSTALL_PROGRAM) -m4755 ./bin/kv4lsetup $(DESTDIR)$(bindir)
 endif	
 ifeq (@USE_QTRECOMP@,1)
-	-$(INSTALL_PROGRAM) ./bin/qtrecompress $(bindir)
+	-$(INSTALL_PROGRAM) ./bin/qtrecompress $(DESTDIR)$(bindir)
 endif
 endif
 ifeq (@USE_JPEGLIB@,1)
 	-$(INSTALL_PROGRAM) ./lib/libavi_mjpeg.so @WIN32_PATH@
 endif
-	-$(INSTALL_PROGRAM) -m0755 ./avifile-config $(bindir)
-	/sbin/ldconfig @LDCONFIGOPTS@
+	-$(INSTALL_PROGRAM) -m0755 ./avifile-config $(DESTDIR)$(bindir)
 
 devinst:
 	-$(INSTALL_PROGRAM) -d $(datadir)/avifile
@@ -117,7 +116,6 @@
 endif
 	-ln -snf `pwd`/avifile-config $(bindir)
 	-chmod 755 $(bindir)/avifile-config
-	/sbin/ldconfig @LDCONFIGOPTS@
 
 uninstall:
 	-rm -fR $(bindir)/aviplay
