diff -urN avifile-0.6.0.orig/configure.in avifile-0.6.0/configure.in
--- avifile-0.6.0.orig/configure.in	Fri Jan 26 17:57:14 2001
+++ avifile-0.6.0/configure.in	Wed Feb 14 16:53:47 2001
@@ -79,11 +79,6 @@
 AC_FUNC_VPRINTF
 AC_CHECK_FUNCS(ftime gettimeofday strdup strstr)
 
-dnl remove me
-CFLAGS="$CFLAGS -march=i586"
-
-
-
 USE_QT=1
 USE_QTRECOMP=1
 
diff -urN avifile-0.6.0.orig/lib/aviplay/Makefile.am avifile-0.6.0/lib/aviplay/Makefile.am
--- avifile-0.6.0.orig/lib/aviplay/Makefile.am	Tue Jan 23 07:16:31 2001
+++ avifile-0.6.0/lib/aviplay/Makefile.am	Wed Feb 14 16:53:47 2001
@@ -1,7 +1,7 @@
 noinst_LTLIBRARIES=libaviplay.la
 libaviplay_la_SOURCES=audio.cpp aviplay.cpp renderer.cpp aviplay2.cpp
 
-CXXFLAGS=-march=i586 -I../../include @SDL_CFLAGS@ @X_INCLUDES@
+CXXFLAGS=-I../../include @SDL_CFLAGS@ @X_INCLUDES@
 
 LIBS=
 noinst_HEADERS=RegAccess.h audio.h aviplay2.h aviplay_impl.h
diff -urN avifile-0.6.0.orig/lib/aviread/Makefile.am avifile-0.6.0/lib/aviread/Makefile.am
--- avifile-0.6.0.orig/lib/aviread/Makefile.am	Tue Jan 23 07:16:15 2001
+++ avifile-0.6.0/lib/aviread/Makefile.am	Wed Feb 14 16:53:47 2001
@@ -1,6 +1,6 @@
 noinst_LTLIBRARIES=libaviread.la
 libaviread_la_SOURCES=asfinput.cpp AsfReadHandler.cpp AviReadHandler.cpp AviRead.cpp Cache.cpp
 noinst_HEADERS=asfinput.h AsfReadHandler.h AviReadHandler.h AviRead.h Cache.h asffmt.h
-CXXFLAGS=-I../../include -Wno-multichar -fno-omit-frame-pointer
+CXXFLAGS=-I../../include -Wno-multichar
 
 LIBS=
diff -urN avifile-0.6.0.orig/lib/common/Makefile.am avifile-0.6.0/lib/common/Makefile.am
--- avifile-0.6.0.orig/lib/common/Makefile.am	Tue Jan 23 07:16:11 2001
+++ avifile-0.6.0/lib/common/Makefile.am	Wed Feb 14 16:53:47 2001
@@ -1,4 +1,4 @@
 noinst_LTLIBRARIES=libcommon.la
 libcommon_la_SOURCES=mmx.cpp image.cpp impl_video.cpp impl_audio.cpp util.cpp ldt_keeper.cpp
-CXXFLAGS=-I../../include -fno-exceptions -Os
+CXXFLAGS=-I../../include -fno-exceptions
 
diff -urN avifile-0.6.0.orig/player/Makefile.am avifile-0.6.0/player/Makefile.am
--- avifile-0.6.0.orig/player/Makefile.am	Tue Jan 23 07:16:42 2001
+++ avifile-0.6.0/player/Makefile.am	Wed Feb 14 16:53:47 2001
@@ -29,7 +29,7 @@
 DEFINES=
 
 CXXFLAGS=-I../include @X_INCLUDES@ @QT_INCLUDES@ @SDL_CFLAGS@ \
-    -DSHARE_PATH=\"@datadir@/avifile\" -DWIN32_PATH=\"@WIN32_PATH@\" -march=i586
+    -DSHARE_PATH=\"@datadir@/avifile\" -DWIN32_PATH=\"@WIN32_PATH@\"
     
 MOC=@MOC@
 UIC=@UIC@
diff -urN avifile-0.6.0.orig/plugins/libmp3lame_audioenc/lame3.70/Makefile.am avifile-0.6.0/plugins/libmp3lame_audioenc/lame3.70/Makefile.am
--- avifile-0.6.0.orig/plugins/libmp3lame_audioenc/lame3.70/Makefile.am	Tue Jan 23 07:15:16 2001
+++ avifile-0.6.0/plugins/libmp3lame_audioenc/lame3.70/Makefile.am	Wed Feb 14 16:53:47 2001
@@ -31,7 +31,7 @@
         interface.c \
         main.c 
 
-CC_OPTS =  -O3 -fomit-frame-pointer -funroll-loops -ffast-math  -finline-functions -Wall
+CC_OPTS = -fomit-frame-pointer -funroll-loops -ffast-math  -finline-functions -Wall
 
 # 10/99 added -D__NO_MATH_INLINES to fix a bug in *all* versions of
 # gcc 2.8+ as of 10/99.  
@@ -44,4 +44,4 @@
 get_audio.h gpkplotting.h gtkanal.h huffman.h id3tag.h ieeefloat.h\
 l3bitstream-pvt.h l3bitstream.h l3side.h lame.h machine.h mpg123.h\
 mpglib.h newmdct.h portableio.h psymodel.h quantize-pvt.h quantize.h\
-reservoir.h rtp.h tables.h timestatus.h util.h version.h 
\ No newline at end of file
+reservoir.h rtp.h tables.h timestatus.h util.h version.h 
diff -urN avifile-0.6.0.orig/plugins/libmpeg_audiodec/Makefile.am avifile-0.6.0/plugins/libmpeg_audiodec/Makefile.am
--- avifile-0.6.0.orig/plugins/libmpeg_audiodec/Makefile.am	Mon Jan 29 12:37:31 2001
+++ avifile-0.6.0/plugins/libmpeg_audiodec/Makefile.am	Wed Feb 14 16:53:47 2001
@@ -4,6 +4,6 @@
  mpegtoraw.cpp
 libmpeg_audiodec_la_LDFLAGS=-module
 noinst_HEADERS=mpegsound.h mpegsound_locals.h fillplugins.h
-CXXFLAGS=-I../../include -O2 -fno-exceptions
+CXXFLAGS=-I../../include -fno-exceptions
 
 LIBS=
diff -urN avifile-0.6.0.orig/plugins/libwin32/loader/Makefile.am avifile-0.6.0/plugins/libwin32/loader/Makefile.am
--- avifile-0.6.0.orig/plugins/libwin32/loader/Makefile.am	Wed Jan 31 20:32:28 2001
+++ avifile-0.6.0/plugins/libwin32/loader/Makefile.am	Wed Feb 14 16:54:29 2001
@@ -4,7 +4,7 @@
 libloader_la_LIBADD=stubs.lo
 noinst_HEADERS=win32.h wineacm.h loader.h
 CFLAGS=-I. -I../../../include -DWIN32_PATH=\"@WIN32_PATH@\" \
--D__WINE__ -Ddbg_printf=__vprintf -DTRACE=__vprintf -fno-omit-frame-pointer -fno-inline -fcaller-saves
+-D__WINE__ -Ddbg_printf=__vprintf -DTRACE=__vprintf -fno-omit-frame-pointer -fcaller-saves
 
 LIBS=
 EXTRA_DIST=stubs.s
