diff -Nur avifile0.7-0.7.7.orig/libavqt/Makefile.am avifile0.7-0.7.7/libavqt/Makefile.am
--- avifile0.7-0.7.7.orig/libavqt/Makefile.am	Wed May 22 22:41:03 2002
+++ avifile0.7-0.7.7/libavqt/Makefile.am	Thu Jun 13 11:01:28 2002
@@ -30,7 +30,7 @@
 
 
 #order.cpp
-libqavm_la_LDFLAGS = $(QT_LIBS) -release $(LT_RELEASE) -version-info $(LT_VERSION)
+libqavm_la_LDFLAGS = $(X_LIBS) $(QT_LIBS) -release $(LT_RELEASE) -version-info $(LT_VERSION)
 #libavqt_la_LIBADD = ../lib/libaviplay.la
 
 #noinst_PROGRAMS = ordertest
