--- avifile-0.7-0.7.38/lib/aviread/FFReadStream.cpp.orig	2003-05-27 17:26:00.000000000 +0200
+++ avifile-0.7-0.7.38/lib/aviread/FFReadStream.cpp	2003-12-30 03:24:43.814350536 +0100
@@ -19,6 +19,7 @@
     { CODEC_ID_H263, mmioFOURCC('H', '2', '6', '3') },
     { CODEC_ID_H263P, mmioFOURCC('H', '2', '6', '3') },
     { CODEC_ID_MP2, 0x50 },
+    { CODEC_ID_MP3LAME, 0x55 },
     { CODEC_ID_AC3, 0x2000 },
     { CODEC_ID_DVVIDEO, mmioFOURCC('D', 'V', 'S', 'D') },
     { CODEC_ID_DVAUDIO, ('D' << 8) | 'A' },
