--- avifile-0.6/configure.in.orig	Mon Oct  1 20:32:07 2001
+++ avifile-0.6/configure.in	Tue Oct  2 00:11:16 2001
@@ -61,6 +61,7 @@
 dnl new automake doesn't like it  
 dnl but its absence breaks old automake ( 1.4 )
 AM_PROG_LIBTOOL
+AM_PROG_AS
 
 dnl Checks for libraries.
 dnl Replace `main' with a function in -ldl:
--- avifile-0.6/plugins/libwin32/loader/Makefile.am.orig	Fri Aug 17 14:49:21 2001
+++ avifile-0.6/plugins/libwin32/loader/Makefile.am	Tue Oct  2 00:18:32 2001
@@ -1,11 +1,11 @@
 noinst_LTLIBRARIES = libloader.la
 libloader_la_SOURCES = afl.c driver.c elfdll.c ext.c module.c \
- pe_image.c pe_resource.c registry.c resource.c win32.c
+ pe_image.c pe_resource.c registry.c resource.c win32.c stubs.s
 
 # ldt_keeper.c is not used here !! this is not a bug 
 # it's used in lib/common/ldt_keeper.cpp
 
-libloader_la_LIBADD = stubs.lo
+#libloader_la_LIBADD = stubs.lo
 noinst_HEADERS = win32.h wineacm.h loader.h ext.h driver.h com.h registry.h ldt_keeper.h
 
 
