diff -urN avifile-0.6.org/lib/aviplay/aviplay_impl.h avifile-0.6/lib/aviplay/aviplay_impl.h
--- avifile-0.6.org/lib/aviplay/aviplay_impl.h	Mon Apr  2 15:53:38 2001
+++ avifile-0.6/lib/aviplay/aviplay_impl.h	Mon Apr  2 15:54:03 2001
@@ -231,7 +231,7 @@
     }
     int Fps() const
     {
-	return (m_pVideostream) ? (int)(1./m_pVideostream->GetFrameTime()) : 0;
+	return (m_pVideostream) ? (int)((1.0/m_pVideostream->GetFrameTime())+0.5) : 0;
     }
     virtual const char* FileName()
     {
