diff -Nuard avifile-0.7-0.7.38.orig/avifile-config.in avifile-0.7-0.7.38/avifile-config.in
--- avifile-0.7-0.7.38.orig/avifile-config.in	2003-05-25 22:44:21.000000000 +0200
+++ avifile-0.7-0.7.38/avifile-config.in	2003-10-18 15:32:54.000000000 +0200
@@ -17,7 +17,7 @@
 data_dir="@datadir@/@PACKAGE@"
 version="@VERSION@"
 
-include_dir="@includedir@/@PACKAGE@"
+include_dir="@includedir@/avifile"
 lib_dir="@libdir@"
 
 usage()
