--- avifile-0.7-0.7.38/plugins/libwin32/loader/ldt_keeper.c.orig	2002-08-08 11:05:51.000000000 +0200
+++ avifile-0.7-0.7.38/plugins/libwin32/loader/ldt_keeper.c	2003-12-29 22:32:11.379726192 +0100
@@ -23,6 +23,7 @@
 #include <unistd.h>
 #ifdef __linux__
 #include <asm/unistd.h>
+#define user_desc modify_ldt_ldt_s
 #include <asm/ldt.h>
 /* prototype it here, so we won't depend on kernel headers */
 #ifdef  __cplusplus
