--- avifile0.6-20011220/plugins/libwin32/loader/win32.c	Sun Jan 20 16:39:53 2002
+++ avifile0.6-20011220/plugins/libwin32/loader/win32.c.new	Sun Jan 20 16:53:14 2002
@@ -2573,7 +2573,7 @@
     local_time=tv.tv_sec;
     local_tm=localtime(&local_time);
 
-    systime->wYear = local_tm->tm_year + 1900;
+    systime->wYear = 1999; /* local_tm->tm_year + 1900; */
     systime->wMonth = local_tm->tm_mon + 1;
     systime->wDayOfWeek = local_tm->tm_wday;
     systime->wDay = local_tm->tm_mday;
@@ -2599,7 +2599,7 @@
     local_time=tv.tv_sec;
     local_tm=gmtime(&local_time);
 
-    systime->wYear = local_tm->tm_year + 1900;
+    systime->wYear = 1999; /* local_tm->tm_year + 1900; */
     systime->wMonth = local_tm->tm_mon + 1;
     systime->wDayOfWeek = local_tm->tm_wday;
     systime->wDay = local_tm->tm_mday;
