diff -urN avifile0.6-20011220.orig/configure.in avifile0.6-20011220/configure.in
--- avifile0.6-20011220.orig/configure.in	Sun Jan 20 17:15:14 2002
+++ avifile0.6-20011220/configure.in	Sun Jan 20 17:17:59 2002
@@ -83,6 +83,7 @@
 dnl automatic updating of libtool configuration files
 dnl not needed with automake AC_SUBST(LIBTOOL_DEPS) 
 dnl using rather hack for .lo creation AM_PROG_AS
+AM_PROG_AS
 
 
 dnl 
diff -urN avifile0.6-20011220.orig/plugins/libwin32/loader/Makefile.am avifile0.6-20011220/plugins/libwin32/loader/Makefile.am
--- avifile0.6-20011220.orig/plugins/libwin32/loader/Makefile.am	Sun Jan 20 17:15:14 2002
+++ avifile0.6-20011220/plugins/libwin32/loader/Makefile.am	Sun Jan 20 17:19:21 2002
@@ -1,7 +1,7 @@
 SUBDIRS = dshow wine 
 noinst_LTLIBRARIES = libloader.la
 libloader_la_SOURCES = afl.c driver.c elfdll.c ext.c module.c \
- pe_image.c pe_resource.c registry.c resource.c win32.c ldt_keeper.c 
+ pe_image.c pe_resource.c registry.c resource.c win32.c ldt_keeper.c stubs.s
 
 # ldt_keeper.c is not used here !! this is not a bug 
 # it's used in lib/common/ldt_keeper.cpp
