--- avifile-0.52/lib/videocodec/Module.cpp~	Sun Dec 17 01:07:43 2000
+++ avifile-0.52/lib/videocodec/Module.cpp	Fri Dec 22 19:07:20 2000
@@ -49,7 +49,7 @@
 	    int newkey;
 	    int bitrate;	
     	    int count=4;
-	    if(RegOpenKeyExA(HKEY_CURRENT_USER, "Software\\LinuxLoader\\div3", 0, 0, &newkey)!=0)
+	    if(RegOpenKeyExA(HKEY_CURRENT_USER, "Software\\LinuxLoader\\Div3", 0, 0, &newkey)!=0)
 	    {
 //		cerr<<"Could not open Divx key"<<endl;
 		goto no_reg;
@@ -61,6 +61,7 @@
 	        goto no_reg;
     	    }    
  
+	    cout << "::module bitrate: " << bitrate << endl;
 	    *(double*)((char*)_handle+0x14c0)=bitrate;
 	    RegCloseKey(newkey);
 	no_reg:
