diff -ur avifile0.7-20020412/include/fourcc.h avifile0.7-20020412.xvid/include/fourcc.h
--- avifile0.7-20020412/include/fourcc.h	Tue Apr  2 11:24:58 2002
+++ avifile0.7-20020412.xvid/include/fourcc.h	Tue Jun 11 01:10:21 2002
@@ -104,7 +104,8 @@
 #define RIFFINFO_ASV2	mmioFOURCC('A', 'S', 'V', '2')
 
 /* let's give some FCC to Xvid */
-#define RIFFINFO_XVID   mmioFOURCC('X', 'v', 'i', 'D')
+#define RIFFINFO_XviD   mmioFOURCC('X', 'v', 'i', 'D')
+#define RIFFINFO_XVID   mmioFOURCC('X', 'V', 'I', 'D')
 
 /* Microsoft video */
 #define RIFFINFO_CRAM	mmioFOURCC('C', 'R', 'A', 'M')
@@ -235,6 +236,7 @@
 #define fccMJPG		RIFFINFO_MJPG
 #define fccmjpg		RIFFINFO_mjpg
 #define fccHFYU		RIFFINFO_HFYU
+#define fccXviD		RIFFINFO_XviD
 #define fccXVID		RIFFINFO_XVID
 #define fccCVID		RIFFINFO_CVID
 #define fcccvid		RIFFINFO_cvid
diff -ur avifile0.7-20020412/plugins/libwin32/fillplugins.h avifile0.7-20020412.xvid/plugins/libwin32/fillplugins.h
--- avifile0.7-20020412/plugins/libwin32/fillplugins.h	Mon Apr  8 09:47:50 2002
+++ avifile0.7-20020412.xvid/plugins/libwin32/fillplugins.h	Tue Jun 11 01:11:41 2002
@@ -401,6 +401,7 @@
         "http://www.geocities.com/avihpit/xvid/post.html";
     static const fourcc_t xvid_codecs[] =
     {
+	fccXviD,
 	fccXVID,
 	fccDIVX, fccdivx, 0
     };
diff -ur avifile0.7-20020412/plugins/libwin32/libwin32.cpp avifile0.7-20020412.xvid/plugins/libwin32/libwin32.cpp
--- avifile0.7-20020412/plugins/libwin32/libwin32.cpp	Thu Apr  4 22:16:07 2002
+++ avifile0.7-20020412.xvid/plugins/libwin32/libwin32.cpp	Tue Jun 11 01:12:17 2002
@@ -284,6 +284,7 @@
 	    || strcmp(attribute, "strSettings") == 0)
 	    return GetRegValue(vp31name, attribute, 0, &value, 0);
         break;
+    case fccXviD:
     case fccXVID:
 	return GetRegValue(xvidname, attribute, 0, &value, 0);
     case fccMJPG:
@@ -370,6 +371,7 @@
 	    || strcmp(attribute, "strSettings") == 0)
 	    return SetRegValue(vp31name, attribute, value, 0);
         break;
+    case fccXviD:
     case fccXVID:
 	return SetRegValue(xvidname, attribute, value, 0);
     case fccMJPG:
diff -ur avifile0.7-20020412/plugins/libwin32/videocodec/VideoDecoder.cpp avifile0.7-20020412.xvid/plugins/libwin32/videocodec/VideoDecoder.cpp
--- avifile0.7-20020412/plugins/libwin32/videocodec/VideoDecoder.cpp	Mon Apr  8 09:47:51 2002
+++ avifile0.7-20020412.xvid/plugins/libwin32/videocodec/VideoDecoder.cpp	Tue Jun 11 01:12:36 2002
@@ -402,6 +402,7 @@
 
     switch (m_Info.fourcc)
     {
+    case fccXviD:
     case fccXVID:
 	if (m_bitrick.biCompression == 0 || m_bitrick.biCompression == 3)
 	    break;
diff -ur avifile0.7-20020412/plugins/libxvid/fillplugins.h avifile0.7-20020412.xvid/plugins/libxvid/fillplugins.h
--- avifile0.7-20020412/plugins/libxvid/fillplugins.h	Mon Apr  8 09:47:51 2002
+++ avifile0.7-20020412.xvid/plugins/libxvid/fillplugins.h	Tue Jun 11 01:11:15 2002
@@ -11,6 +11,7 @@
     static char xvid_about[] = "XviD Linux codec";
     static const fourcc_t xvid_codecs[] =
     {
+        fccXviD, 
         fccXVID, 
 	fccDIVX, fccdivx, 0
     };
