diff -uNr avifile0.7-0.7.16.orig/lib/video/renderer.cpp avifile0.7-0.7.16/lib/video/renderer.cpp
--- avifile0.7-0.7.16.orig/lib/video/renderer.cpp	Thu Sep 12 20:52:51 2002
+++ avifile0.7-0.7.16/lib/video/renderer.cpp	Wed Oct  2 21:02:00 2002
@@ -2766,11 +2766,13 @@
 				 int width, int height,
 				 fourcc_t yuv_fmt, bool sub)
 {
+#ifdef HAVE_VIDIX
     //return new XvYUVRenderer(pw, dpy, width, height, yuv_fmt, sub);
     if (geteuid() == 0)
 	// for root user try to use VIDIX
 	try { return new VidixRenderer(pw, (Display*)dpy, width, height, yuv_fmt, sub);
 	} catch (FatalError& e)  { e.PrintAll(); }
+#endif
 #if _SDL_VER > 1103
     return new YUVRenderer(pw, (Display*)dpy, width, height, yuv_fmt, sub);
 #else
